### DISCUSSION TITLE
Get corresponding VTK data object from a Dataset

### DISCUSSION THREAD

**Discussion by njneeteson** [pyvista/pyvista#2190] [Q&A] [answered]
The [docs page on transitioning from VTK to PyVista ](https://docs.pyvista.org/user-guide/vtk_to_pyvista.html) has this line:

> In this manner, you can get the “best of both worlds” should you need the flexibility of PyVista and the functionality of VTK.

I can imagine use cases where I would want to switch back and forth between PyVista and VTK (or maybe in some stage of my workflow I want to go PyVista->VTK->SITK->VTK->PyVista) to take advantage of as-of-yet unwrapped filters available in VTK (or SITK, or other VTK-data-compatible libraries/packages). However, while wrapping a VTK data object into a PyVista Dataset is well-documented, I have not been able to find anything about going from a PyVista Dataset back to a corresponding VTK data object (e.g. get a vtkImageData from a UniformGrid with the origin, spacing, scalar data, and field data / "header data" preserved as much as possible).

Is this implemented and I just can't find it? Is this a feature that is planned for the future? 

Or, is the purpose of PyVista more for visualization, and in my workflow I should front-load all of my VTK/SITK operations and only wrap my data into a PyVista Dataset when I'm ready to visualize or do other tasks that can be done entirely within PyVista?

**Comment by whophil** [comment, marked as answer]
If I'm understanding your question correctly, you can do this today. The PyVista `UniformGrid` object _is_ indeed a `vtkImageData`, and you can pass it to any function or filter which accepts a `vtkImageData`, or call any methods that `vtkImageData` provides.

**Reply by njneeteson** [reply]
Thank you very much! Is this documented somewhere and I just missed it? If not it should have more emphasis given to it particularly in the parts of the documentation that seem designed to "convert" VTK users - this is very powerful and makes this library much more useful than I already though it would be in my image processing/analysis workflows.

**Reply by whophil** [reply]
@njneeteson I looked through the docs and was surprised to find that it not well documented!

CC: @akaszynski @banesullivan @tkoyama010

**Reply by RichardScottOZ** [reply]
Possibly a good example case this one then.

**Reply by tkoyama010** [reply]
What we should tell you is that pyvista classes inherit from vtk classes, it would be nice to know the correspondence between which pyvista classes inherit from which vtk classes. Of course, it is also important to provide examples.
